### ISSUE TITLE
Internal Server Error: esbuild_1.startService is not a function

### ISSUE THREAD

**Issue by khalibloo** [vitejs/vite-plugin-react-pages#39]
While trying to reproduce #37 I stumbled upon this one. Here is the reproduction for it https://github.com/khalibloo/vite-pages-esbuild-bug
In both issues, I used the same commands on 2 different machines at the same time, and got different project scaffolding. Comparing the package.json alone shows they're very different. The machine on #37 is on node v14.17.4 and npm v6.14.14, while the one for this issue is on node v14.17.1 and npm v6.14.13. No significant version differences.

Steps (use pnpm or npm, both give the same error)
1. Create a new vite-pages project `pnpm init vite-pages app-demo --template app`
2. Install dependencies `pnpm i`
3. Start dev server `pnpm dev`

Error message
```
> pnpm dev

> vite-pages-demo@1.0.2 dev D:\Workspace\Web\Playground\vite-pages-demo
> vite serve

(!) "alias" option is deprecated. Use "resolve.alias" instead.
Pre-bundling dependencies:
  react
  react-dom
  react-router-dom
(this will be run only when your dependencies or config have changed)

  vite v2.5.6 dev server running at:

  > Local: http://localhost:3000/
  > Network: use `--host` to expose

  ready in 1044ms.

23:47:29 [vite] new dependencies found: vite-pages-theme-basic, updating...
23:47:29 [vite] ✨ dependencies updated, reloading page...
23:47:30 [vite] Internal server error: esbuild_1.startService is not a function
  Plugin: vite-plugin-mdx
  File: D:/Workspace/Web/Playground/vite-pages-demo/pages/$.mdx
      at D:\Workspace\Web\Playground\vite-pages-demo\node_modules\.pnpm\vite-plugin-mdx@2.0.1_react@17.0.2+vite@2.5.6\node_modules\vite-plugin-mdx\dist\transform.js:47:34
      at Generator.next (<anonymous>)
      at D:\Workspace\Web\Playground\vite-pages-demo\node_modules\.pnpm\vite-plugin-mdx@2.0.1_react@17.0.2+vite@2.5.6\node_modules\vite-plugin-mdx\dist\transform.js:8:71      at new Promise (<anonymous>)
      at __awaiter (D:\Workspace\Web\Playground\vite-pages-demo\node_modules\.pnpm\vite-plugin-mdx@2.0.1_react@17.0.2+vite@2.5.6\node_modules\vite-plugin-mdx\dist\transform.js:4:12)
      at ensureEsbuildService (D:\Workspace\Web\Playground\vite-pages-demo\node_modules\.pnpm\vite-plugin-mdx@2.0.1_react@17.0.2+vite@2.5.6\node_modules\vite-plugin-mdx\dist\transform.js:45:12)
      at Object.<anonymous> (D:\Workspace\Web\Playground\vite-pages-demo\node_modules\.pnpm\vite-plugin-mdx@2.0.1_react@17.0.2+vite@2.5.6\node_modules\vite-plugin-mdx\dist\transform.js:25:31)
      at Generator.next (<anonymous>)
      at fulfilled (D:\Workspace\Web\Playground\vite-pages-demo\node_modules\.pnpm\vite-plugin-mdx@2.0.1_react@17.0.2+vite@2.5.6\node_modules\vite-plugin-mdx\dist\transform.js:5:58)
```

**Comment by csr632** [comment]
It is fixed in the latest version. Could you try again?

**Comment by khalibloo** [comment]
On the machine that produces this issue, creating a new vite-pages project still gives me a package.json like this https://github.com/khalibloo/vite-pages-esbuild-bug/blob/master/package.json with vite-plugin-react-pages version 2.x.
Any ideas why running the npm init would give me the old version of the template? I believe the init command should always fetch the latest template, right?

**Comment by khalibloo** [comment]
I ran `yarn create vite-pages app-demo --template app`, which got the latest create-vite-pages. Then npm and pnpm just fixed themselves right after, and started fetching the latest version as well. I have no clue what caused the issue or why using yarn fixed npm and pnpm. It's also a bit concerning that npm init isn't guaranteed to fetch the latest versions of templates, but at least I'm unstuck for now. Thanks for your help!

**Comment by csr632** [comment]
It may be a bug of npm's cache. (By the way, are you using a custom npm regiatery?)

Try upgrading your npm version. Or delete npm cache.
This may be helpful: https://docs.npmjs.com/cli/v7/commands/npm-exec#a-note-on-caching

**Comment by khalibloo** [comment]
Nope, default configs for all 3 package managers. I suspected it might have been a cache issue with npm initially, given that i've used vite-pages on the machine with the issue a long time ago. But I just installed pnpm maybe 2 days ago. It didn't make sense that pnpm would also have the old create-vite-pages in its cache, unless it's somehow also using npm's cache behind the scenes. Oh and npm assured me that i shouldn't need to clean its cache when I tried to. So I quickly wrote that off.

Then I figured maybe it's the registry, and since yarn uses a different registry, that might work. And it did. Then I went back to npm to compare the outputs again and it also just worked. And so did pnpm. But I force-cleaned its cache anyway just for good measure.
